phoenix-core/src/main/java/org/apache/phoenix/compile/WhereOptimizer.java

```diff
@@ -242,6 +260,12 @@ public static Expression pushKeyExpressionsToScan(StatementContext context, Filt
                     cnf.add(leftRanges);
                     clipLeftSpan = 0;
                     prevSortOrder = sortOrder;
+                    // since we have to clip the portion with the same sort order, we can no longer
+                    // extract the nodes from the where clause
+                    // for eg. for the schema A VARCHAR DESC, B VARCHAR ASC and query WHERE (A,B) < ('a','b')
+                    // the range (* - a\xFFb) is converted to (~a-*)(*-b)
+                    // so we still need to filter on A
+                    stopExtracting = true;
```

Comment:
Since you added this new scenario setting stopExtracting to true. Can you also update the comment below at line 290:"// Stop extracting nodes once we encounter" accordingly?

Reply:
Ok